### ISSUE TITLE
Build attempt fails on Windows

### ISSUE THREAD

**Issue by ElectricRCAircraftGuy** [ElectricRCAircraftGuy/Microchip_XC32_Compiler#1]
See my Q here: [`checking size of mp_limb_t... 0` and `configure: error: Oops, mp_limb_t doesn't seem to work` when compiling the Microchip XC32 compiler from source](https://stackoverflow.com/q/77414776/4561887), especially the "**More leads to follow up on**" section at the bottom of the question.

**Comment by ElectricRCAircraftGuy** [comment]
I answered the above question. Here's the next build error I'm stuck on: [[C++: `error: expected ',' or '...' before string constant` before `__FILE__` in `fancy_abort (__FILE__, __LINE__, __FUNCTION__)` when compiling gcc](https://stackoverflow.com/questions/77435355/c-error-expected-or-before-string-constant-before-file-in)](https://stackoverflow.com/q/77435355/4561887)

**Comment by ElectricRCAircraftGuy** [comment]
It works. Both Q's above are solved.

**Comment by ElectricRCAircraftGuy** [comment]
See my bug report here, too: https://github.com/msys2/MINGW-packages/issues/19036

**Comment by oleid** [comment]
Did you ever get this working?
I managed to build  the compiler, too, however, when using it it didn't find the system headers. It would appear that the default include paths were incorrect.

Furthermore, I managed to get a canadian cross build of the compiler going, however, here the linker complains about different floating point ABI - same options work for the official build.

The only thing which really works for me is the linux build of the linux compiler.

**Comment by ElectricRCAircraftGuy** [comment]
> Did you ever get this working? 

@oleid , yes, I did get it fully building and compiling the full compiler in Windows (and in Linux, which was easier). 

But, that's where I stopped. I did not attempt to build code with the compiler I compiled. Do you have the compiler fully building in Windows?

**Comment by ElectricRCAircraftGuy** [comment]
My notes are an absolute mess. It was so brutally hard getting it to build then I never got time to come back and clean up my notes. But, this is the build script that ended up working for me in Windows: https://github.com/ElectricRCAircraftGuy/Microchip_XC32_Compiler/blob/main/wip_Windows_build_scripts/build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh

**Comment by oleid** [comment]
Thank you very much! 
Yes, I got it building - both, the Canadian cross compiler and the msys based build thanks to your notes. The Canadian cross, however, is strangely broken (as I mentioned before), the msys based compiler looks best. The way I see it the default include search paths are broken. If I get it fixed I'll report back to you.

**Comment by Nikolaev-Nikolay** [comment]
```bash
build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh: line 52: make: command not found

real    0m0.143s
user    0m0.000s
sys     0m0.045s
Error: [libmchp] failed to build!
```

Not Working :(
Please email me so that I can download the pre-built compiler from the Pic32. sources. I am unable to compile under Windows.

**Comment by ElectricRCAircraftGuy** [comment]
> build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh: line 52: make: command not found
> 
> real    0m0.143s
> user    0m0.000s
> sys     0m0.045s
> Error: [libmchp] failed to build!

Looks like you're missing `make`. In your ucrt64 terminal you set up by following [my MSYS2 installation instructions](https://stackoverflow.com/a/77407282/4561887) just above, run this to install `make`: 
```bash
pacman -S mingw-w64-ucrt-x86_64-make    # Install make
```
See my answer here for where I got that: https://stackoverflow.com/a/79228659/4561887

Install all of these, actually: 
```bash
pacman -Syu  # run repeatedly until there are no more updates
pacman -S mingw-w64-ucrt-x86_64-gcc     # Install gcc
pacman -S mingw-w64-ucrt-x86_64-cmake   # Install CMake
pacman -S mingw-w64-ucrt-x86_64-make    # Install make
```

I don't have the binaries easily accessible anymore to give to you.

**Comment by Nikolaev-Nikolay** [comment]
Hello!
./build-xc32-v4.35m.sh: line 118: /c/GCC_PIC/Microchip_XC32_Compiler/wip_Windows_build_scripts/xc32-v4.35-src/pic32m-build/opt/include/host-defs.h: No such file or directory

And...
./build-xc32-v4.35m.sh: line 158: /c/GCC_PIC/Microchip_XC32_Compiler/wip_Windows_build_scripts/xc32-v4.35-src/pic32m-source/binutils/configure: No such file or directory
Error: [binutils] failed to build!

real    0m2.287s
user    0m0.075s
sys     0m0.167s

What should I do next with these errors?! Please help me!

<img width="1148" height="524" alt="Image" src="https://github.com/user-attachments/assets/d7bcf28e-5673-478a-b354-64802937b6dc" />

**Comment by ElectricRCAircraftGuy** [comment]
@Nikolaev-Nikolay , perhaps you need to move [`build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh`](https://github.com/ElectricRCAircraftGuy/Microchip_XC32_Compiler/blob/main/wip_Windows_build_scripts/build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh) up one level before running it. It seems to think that the source code is in the `wip_Windows_build_scripts` dir. It isn't. It's up one level from there.

Or, try changing line 5: https://github.com/ElectricRCAircraftGuy/Microchip_XC32_Compiler/blob/main/wip_Windows_build_scripts/build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh#L5C1-L5C53

from: 
```bash
SCRIPT_DIRECTORY="$(dirname "$FULL_PATH_TO_SCRIPT")"
```

to:
```bash
SCRIPT_DIRECTORY="../$(dirname "$FULL_PATH_TO_SCRIPT")"
```

**Comment by Nikolaev-Nikolay** [comment]
O, Yes!

<img width="1366" height="728" alt="Image" src="https://github.com/user-attachments/assets/a8c922db-1420-411a-b3ae-23920f117805" />
Oh, No...
make[1]: Leaving directory '/c/GCC_Pic/Microchip_XC32_Compiler/xc32-v4.35-src/pic32m-build/binutils'
make: *** [Makefile:865: all] Error 2

real    4m31.041s
user    3m5.096s
sys     7m13.767s
Error: [binutils] failed to build!

real    6m13.085s
user    3m33.040s
sys     8m7.002s
Please tell me what this means. I started compiling and this happened. What can I do? I moved the script to the next level. What is the command to run the script? Maybe I'm doing something wrong.

**Comment by ElectricRCAircraftGuy** [comment]
I'm going to need the full output. Paste it into a backtick code block:
````
```
paste your output here
```
````
...or upload a text file of the output. 

With the script inside the `path/to/Microchip_XC32_Compiler/` dir, the command to run the script is simply:
```bash
cd path/to/Microchip_XC32_Compiler
./build-xc32-v4.35m_relative_paths_WORKED_ON_WINDOWS.sh
```

Your Makefile failed at line 865, based on the bottom of your output. What's on that line?

**Comment by Nikolaev-Nikolay** [comment]
Hello! This is output file.

[output.txt.txt](https://github.com/user-attachments/files/26286603/output.txt.txt)
And make file this:

[Makefile.txt](https://github.com/user-attachments/files/26286750/Makefile.txt)
And point 856 This is:

<img width="1366" height="728" alt="Image" src="https://github.com/user-attachments/assets/5e90ca7e-3795-49cd-ae64-4e6fff7828a0" />
